packages/block-library/src/navigation-link/shared/use-entity-binding.js

```diff
@@ -4,6 +4,29 @@
 import { useCallback } from '@wordpress/element';
 import { useBlockBindingsUtils } from '@wordpress/block-editor';
 
+/**
+ * Builds entity binding configuration for navigation link URLs.
+ * This function generates the structure used to bind navigation link URLs to their entity sources.
+ *
+ * Using a function instead of a constant allows for future enhancements where the binding
+ * might need dynamic data (e.g., entity ID, context-specific arguments).
+ *
+ * @param {string} kind - The kind of entity ('post-type' or 'taxonomy')
+ * @return {Object} Entity binding configuration object
+ */
+export function buildNavigationLinkEntityBinding( kind ) {
+	const source = kind === 'taxonomy' ? 'core/term-data' : 'core/post-data';
```

Comment:
Should we only support `post-type` or `taxonomy` here? Right now I could pass some CPT and instead of erroring it would just use the post data binding.